tests/functional/test_cliargumentname_config.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from jsonschema import Draft4Validator
+
+import botocore.session
+from botocore import xform_name
+
+
+SERVICE_SCHEMA = {
```

Comment:
Just realizing this, it may make sense to hoist this to the CLI functional tests (depending on what direction we go as it relates to the file the cliArgumentRenames). Mainly if the plan is to put these in the `cli-extras.json`, botocore will not automatically load the up because botocore will only look for `sdk-extras.json` and therefore from it's standpoint, it will never see the `cliArgumentName`. If it were in the `cli-extras.json` (assuming we add support for that extension in the CLI), the CLI will be able to pick up the cliArgumentRenames in a CLI driver's session.

However if the values are going to be part of the `service-2.json` and not in the `extras.json`, this will not be a problem and the tests can live in botocore or the CLI. But if we put it in the CLI functional tests, it will not matter if the final resting place is the `cli-extra.json` or `service-2.json`. It will be loaded no matter what for the tests.

Let me know what you think.

Reply:
I'm happy having these tests live in either place, and I agree that if we put them in ``cli-extras.json`` we probably don't want the SDK to be loading them by default. That does mean this is blocked on that decision, unless we commit to here for now, and if need be hoist it out later (but we may want to hold off on merging this until we have a decision on that anyways).

Reply:
Yeah I think waiting to make changes/merge this PR until a decision is made makes sense to me.